hist/hbook/inc/THbookBranch.h

```diff
@@ -30,8 +30,8 @@ class THbookBranch : public TBranch {
 
 public:
    THbookBranch() {;}
-   THbookBranch(TTree *tree, const char *name, void *address, const char *leaflist, Int_t basketsize=32000, Int_t compress=-1);
-   THbookBranch(TBranch *branch, const char *name, void *address, const char *leaflist, Int_t basketsize=32000, Int_t compress=-1);
+   THbookBranch(TTree *tree, const char *name, void *address, const char *leaflist, Int_t basketsize=32000, Int_t compress = ROOT::kUseGeneralPurposeCompressionSetting);
```

Comment:
The previous value (-1) means "use the compression level of the associtated file if any which is different from the 'default'.  In TBranch, kUnsetCompressionAlgorithm was used in this PR, which leads to the right behavior but the name of the constant does not clearly indicate the behavior.